server/controllers.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+func HandleSecurityLogin(app App) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		defer func() {
+			if r := recover(); r != nil {
+				http.Error(w, "Unauthorized!", http.StatusUnauthorized)
+				fmt.Println(r)
+			}
+		}()
+
+		dto := LoginPostDTO{}
+		err := json.NewDecoder(req.Body).Decode(&dto)
+		if err != nil {
+			panic("Error deserializing DTO")
```

Comment:
A requirement of the challenge is to not panic in non-critical conditions. Can you think of a better way to handle errors here?

Reply:
Hi @russjones  , I believe an area of focus is for the system to not crash under non-critical errors? This shouldn't crash the system if the JSON deserialization fails but result in an unauthorized response being returned(that's the intention anyway).